src/Processors/Transforms/SquashingChunksTransform.cpp

```diff
@@ -64,8 +69,22 @@ void SimpleSquashingChunksTransform::transform(Chunk & chunk)
     }
     else
     {
+        if (chunk.hasRows())
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Chunk expected to be empty, otherwise it will be lost");
```

Comment:
Don't know if it's important, but new code doesn't fit comment to `SimpleSquashingChunksTransform` - https://github.com/ClickHouse/ClickHouse/blob/ad385238da24682e02e9867b28a38fb060eba834/src/Processors/Transforms/SquashingChunksTransform.h#L31

Reply:
this comment describes different situation (that is not a bug if we consider it in isolation from squashing transform). when the transform reading output from squashing stops accepting chunks while there are some more to be returned from squashing.